src/runtime/markdown-parser/handler/link.ts

```diff
@@ -30,9 +30,9 @@ export default function link (h: H, node: Node) {
 function normalizeLink (link: string) {
   if (isRelative(link) || (!/^https?/.test(link) && !link.startsWith('/'))) {
     return link.split('/')
-      .map(x => x.replace(/^[0-9]*\./g, ''))
+      .map(x => x.replace(/^[0-9]+\./, ''))
```

Comment:
What about ./2.dir/1.file.md ?

Could we add tests for this feature?

Reply:
Since split link by `/` and process each part separately `./2.dir/1.file.md` will not have any issue.

I will add tests for it